packages/@aws-cdk/aws-logs-destinations/test/kinesisfirehose.test.ts

```diff
@@ -0,0 +1,161 @@
+  });
+
+  // THEN: subscription target is firehose
+  expect(stack).toHaveResource('AWS::Logs::SubscriptionFilter', {
+    DestinationArn: { 'Fn::GetAtt': ['MyFirehose', 'Arn'] },
+    RoleArn: { 'Fn::GetAtt': ['SubscriptionCloudWatchLogsCanPutRecordsIntoKinesisFirehoseC45D3D89', 'Arn'] },
+  });
+
+  // THEN: we have a role to write to the firehose
+  expect(stack).toHaveResource('AWS::IAM::Role', {
+    AssumeRolePolicyDocument: {
+      Version: '2012-10-17',
+      Statement: [{
+        Action: 'sts:AssumeRole',
+        Effect: 'Allow',
+        Principal: {
+          Service: {
+            'Fn::Join': ['', [
+              'logs.',
+              { Ref: 'AWS::Region' },
+              '.amazonaws.com',
+            ]],
+          },
+        },
+      }],
+    },
+  });
+
+  expect(stack).toHaveResource('AWS::IAM::Policy', {
+    PolicyDocument: {
+      Version: '2012-10-17',
+      Statement: [
+        {
+          Action: [
+            'firehose:PutRecord',
+            'firehose:PutRecordBatch',
+          ],
+          Effect: 'Allow',
+          Resource: { 'Fn::GetAtt': ['MyFirehose', 'Arn'] },
+        },
+        {
+          Action: 'iam:PassRole',
+          Effect: 'Allow',
+          Resource: { 'Fn::GetAtt': ['SubscriptionCloudWatchLogsCanPutRecordsIntoKinesisFirehoseC45D3D89', 'Arn'] },
+        },
+      ],
+    },
+  });
+});
+
+test('firehose can be subscription destination twice, without duplicating permissions', () => {
+  // GIVEN
+  const stack = new cdk.Stack();
+  const deliveryStream = new firehose.CfnDeliveryStream(stack, 'MyFirehose');
+  const firehoseDestination = new dests.KinesisFirehoseDestination(deliveryStream);
+  const logGroup1 = new logs.LogGroup(stack, 'LogGroup1');
+  const logGroup2 = new logs.LogGroup(stack, 'LogGroup2');
+
+  // WHEN
+  new logs.SubscriptionFilter(stack, 'Subscription1', {
+    logGroup: logGroup1,
+    destination: firehoseDestination,
+    filterPattern: logs.FilterPattern.allEvents(),
+  });
+  new logs.SubscriptionFilter(stack, 'Subscription2', {
+    logGroup: logGroup2,
+    destination: firehoseDestination,
+    filterPattern: logs.FilterPattern.allEvents(),
+  });
+
+  // THEN: subscription target is firehose
+  expect(stack).toHaveResource('AWS::Logs::SubscriptionFilter', {
+    LogGroupName: { Ref: 'LogGroup106AAD846' },
+    DestinationArn: { 'Fn::GetAtt': ['MyFirehose', 'Arn'] },
+    RoleArn: { 'Fn::GetAtt': ['Subscription1CloudWatchLogsCanPutRecordsIntoKinesisFirehoseEAD271F9', 'Arn'] },
+  });
+  expect(stack).toHaveResource('AWS::Logs::SubscriptionFilter', {
+    LogGroupName: { Ref: 'LogGroup2477F707C' },
+    DestinationArn: { 'Fn::GetAtt': ['MyFirehose', 'Arn'] },
+    RoleArn: { 'Fn::GetAtt': ['Subscription2CloudWatchLogsCanPutRecordsIntoKinesisFirehoseB0757929', 'Arn'] },
+  });
+
+  // THEN: we have a single role to write to the Firehose
+  expect(stack).toCountResources('AWS::IAM::Role', 2); // TODO: Fix to 1 - somehow it still creates 2 roles...
```

Comment:
Please change this to 1 so the build will fail until this has been fixed.

Reply:
Any tips on what might be the issue here? I have no idea...

Reply:
But in my first revision it was not working when I used scope of `Stack`.

My initial intention was to have a single role for each Firehose destination. (i.e. 2 log groups will re-use the same role). What scope is the correct one to use for this role?